### ISSUE TITLE
darknet is not using all the available GPU

### ISSUE THREAD

**Issue by mbayog02** [AlexeyAB/darknet#395]
Hi

I am trying to improve darknet framerate because i only get 12/13 fps in a Quadro k2200 with 4Gb
I have make a filter like explained here #229  to only detect a few classes and implement a "dont_show" option to not display the output video and only save it

While using `./darknet detector demo cfg/coco.data cfg/yolo.cfg yolo.weights data/input.mp4 -out_filename out.avi -dont_show` I have realised that nvidia-smi shows darknet only uses 804Mb although the GPU has almost 3Gb available

![screenshot from 2018-02-16 14-01-00](https://user-images.githubusercontent.com/17453296/36308769-4a002918-1322-11e8-8db5-acd4008fc16b.png)

Is it possible to improve the fps rate by allowing darknet to use more GPU?
How could I do this?

Thank you

**Comment by AlexeyAB** [comment]
Hi,

Look at the **GPU-Util  97%** in the nvidia-smi output. It shows how much performance of GPU is used.

For detection much GPU-memory isn't required, it required only for training.

**Comment by mbayog02** [comment]
Ok, I now see it. I haven´t realised that index.

How could I improve my fps rate as much as possible?
I have tried tiny-yolo-voc but the precission is not enought for me.
I have also tried to use tiny-yolo (The coco version) but I have not gotten it to work ( I tried  `./darknet detector demo cfg/coco.data cfg/tiny-yolo.cfg tiny-yolo.weights data/input.mp4 -out_filename out.avi` but I dont get any label in my output file)

Thank you so much

**Comment by VanitarNordic** [comment]
@AlexeyAB 

I don't want to open a new issue about this questions, but may I ask you update the readme about your new updates about mAP calculations? Thanks.

**Comment by AlexeyAB** [comment]
@mbayog02 I just run `darknet.exe detector demo data/coco.data tiny-yolo.cfg tiny-yolo.weights test.mp4 -i 0` on Windows and it works well. I'll try tiny-coco on Linux.
* What GPU do you use?
* Do you see labels on the screen?
* Did you compile with GPU and CUDNN?

**Comment by AlexeyAB** [comment]
@VanitarNordic I added about mAP.

**Comment by mbayog02** [comment]
@AlexeyAB 
I am using an Nvidia Quadro K2200 in a Desktop PC and a Nvidia Jetson TX2. In both I have the same result.
I don´t see any label in the screen, just the original video.
I compiled it with GPU but not with CUDNN. What this excatly means? Is there a big difference by using it?

**Comment by AlexeyAB** [comment]
* Do you use this repository or original?
* What CUDA-version,OpenCV-version and Linux-version do you use?
* Try to download again: 
    * https://pjreddie.com/media/files/tiny-yolo.weights
    * https://raw.githubusercontent.com/pjreddie/darknet/master/cfg/tiny-yolo.cfg

I just copied video to data/input.mp4, then ran `./darknet detector demo cfg/coco.data cfg/tiny-yolo.cfg tiny-yolo.weights data/input.mp4 -out_filename out.avi` on the Linux and got good results - video with detections.
Makefile:
```
GPU=1
CUDNN=0
OPENCV=1
DEBUG=0
OPENMP=0
LIBSO=1
```

Your Quadro K2200 has CC 3.0 so it should work. But try to compile with `CUDNN=1`, cuDNN should be in `/usr/local/cudnn`, download cudnn for your CUDA version: https://developer.nvidia.com/cudnn

**Comment by AlexeyAB** [comment]
Also does the full COCO model work well? `./darknet detector demo cfg/coco.data cfg/yolo.cfg yolo.weights data/input.mp4 -out_filename out.avi`

**Comment by mbayog02** [comment]
I have just tried it again using cudnn and I still don´t get any output labels.

Do you use this repository or original?
 -I have tried both of them
What CUDA-version,OpenCV-version and Linux-version do you use?
 -CUDA 9.0
 -Open CV 3.4.0

The COCO model works well despite the FPS rate.
Is there other way to improve the performance by not reducing the precision?

Thanks

**Comment by AlexeyAB** [comment]
> Is there other way to improve the performance by not reducing the precision?
No, there is no way. There is only trade off between performance and accuracy.

**Comment by mbayog02** [comment]
Ok, Thanks
